src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,550 @@
+        http:Request request = new;
+        json payload = {"name": worksheetName};
+        request.setJsonPayload(payload);
+        http:Response|error httpResponse = self.workbookClient->post("/v1.0/me/drive/root:" +
+                    self.properties.path + self.properties.workbookName + ".xlsx:/workbook/worksheets", request);
+        int position = -1;
+        string sheetId = "";
+
+        if (httpResponse is http:Response) {
+            if (httpResponse.statusCode == HTTP_STATUS_CREATED) {
+                json|error response = httpResponse.getJsonPayload();
+                if (response is map<json>) {
+                    json sheetIdItem = response["id"];
+                    sheetId = sheetIdItem.toString();
+                    json positionItem = response["position"];
+
+                    int|error res1 = ints:fromString(positionItem.toString());
+                    if (res1 is int) {
+                        position = res1;
+                    } else {
+                        error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while " +
+                                                                            "getting the worksheet position.");
+                        return err;
+                    }
+
+                    Worksheet workSheet = new(self.microsoftGraphConfig, self.properties.path,
+                                            self.properties.workbookName, sheetId, worksheetName, position);
+                    return workSheet;
+                } else {
+                    error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while creating the worksheet.");
+                    return err;
+                }
+            } else {
+                error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while creating the worksheet.");
+                return err;
+            }
+        } else {
+            error err = error(WORKSHEET_ERROR_CODE, message = "Error occurred while accessing the Microsoft Graph API.");
+            return err;
+        }
+    }
+
+    # Remove a worksheet from this workbook.
+    # + worksheetName - name of the worksheet to be removed
+    # + return - boolean true on success, else returns an error
+    public remote function removeWorksheet(string worksheetName) returns @tainted (boolean|error) {
+        boolean result = false;
+        http:Request request = new;
+        http:Response|error httpResponse = self.workbookClient->delete("/v1.0/me/drive/root:/" +
+            self.properties.workbookName  + ".xlsx:/workbook/worksheets/" + worksheetName, request);
+
+        if (httpResponse is http:Response) {
+            if (httpResponse.statusCode == HTTP_STATUS_NO_CONTENT) {
+                result = true;
+            } else {
+                error err = error(WORKSHEET_ERROR_CODE, message = "Error occurred while deleting the worksheet.");
+                return err;
```

Comment:
We should be able to do the following everywhere.

```suggestion
                return error(WORKSHEET_ERROR_CODE, message = "Error occurred while deleting the worksheet.");
```

Reply:
I have improved the error handling code. Now the above code segment has been changed as follows.

```
                HttpResponseHandlingError httpResponseHandlingError = error(HTTP_RESPONSE_HANDLING_ERROR,
                message = "Error occurred while deleting the worksheet.", errorCode = HTTP_RESPONSE_HANDLING_ERROR);
                return httpResponseHandlingError;
```